src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java

```diff
@@ -89,7 +90,8 @@
    * @return  an instance of the KafkaReadStream
    */
   static <K, V> KafkaReadStream<K, V> create(Vertx vertx, Properties config) {
-    return create(vertx, new org.apache.kafka.clients.consumer.KafkaConsumer<>(config));
+    ConsumerTracer tracer = ConsumerTracer.create(vertx, config::getProperty);
+    return new KafkaReadStreamImpl<>(vertx.getOrCreateContext(), new org.apache.kafka.clients.consumer.KafkaConsumer<>(config), tracer);
```

Comment:
as far as I understood the tracing will be always enabled because you are creating a non null tracer in each create method. How people can avoid having tracing then? The same comment for the write stream implementation.

Reply:
The `create` can return null (I annotated @Nullable to make it more explicit), it does when tracing isn't enabled globally in vertx: see https://github.com/vert-x3/vertx-kafka-client/pull/185/files#diff-10d8b5db3bb9723f7d57a3532fbc52e0R30-R33

This is quite similar to how it's done in other client such as SQL.

Perhaps I can rename to `createIfEnabled`, or something like that, to make it more explicit?

Reply:
No no, that's fine. Never used the Vert.x tracing stuff so I didn't get how it was enabled. Btw is it something from Vert.x 4 only right?

Reply:
I think it's been in 4.0.0-SNAPSHOT since its creation last year, not sure if it was in 3.x branch also. I'm using it for demo purpose so sticking to the very latest :)

Reply:
Well, not many people will move to Vert.x 4 so if tracing is even in 3.9.x you should think about cherry-picking of having another PR for it.

Reply:
Tracing is a Vert.x 4 feature that cannot be back-ported